### DISCUSSION TITLE
请问如何能测试全国其他地区比较友好的ip呢？

### DISCUSSION THREAD

**Discussion by nj-dream** [XIU2/CloudflareSpeedTest#338] [讨论]
你好，通过程序可以测试对于本地友好的cf ip，如果想让网站在使用cf后，让全国的网站访问者都能有良好的速度，如何能针对不同地区找到优化的ip呢？谢谢！

**Comment by XIU2** [comment]
那你就需要分别在**电信、联通、移动**网络环境下测速得到 3 个最快 IP。

**Comment by nj-dream** [comment]
没想到这么快得到回复，多谢，多谢！如果我在本地，分别通过电信、移动、联通网络测试，得到的最快3个ip，那么，国内其他地区相同电信服务商的用户，访问时的体验，应该也差异不大吧。

**Comment by XIU2** [comment]
理论上只有在当前网络环境下测速才能得到最适合当前的 IP。
不过对于网站自选来说显然不可能实现，所以只能退而求其次。

同一个 IP 不同运营商下天差地别，同一个运营商下不同地区也存在差异，但是相比默认分配的 IP，还是要好点的，起码不会更差~

**Comment by nj-dream** [comment]
哈哈，明白了，多谢多谢！

**Comment by XIU2** [comment]
做网站的话，可以省去下载测速，只需要延迟测速就够了。
**提高延迟测速次数** 及 **三网测速频率**，这个比较重要，毕竟是 Anycast IP，不知道什么时候会变化。

当然如果你要省力的话，可以直接用隔壁项目现成的方案：
https://github.com/ddgth/cf2dns
https://stock.hostmonit.com/CloudFlareYes

**Comment by nj-dream** [comment]
太好了，正在找相关的资料呢。我想在网站上放视频来讲解产品。但使用国内的cdn，担心被攻击，一个晚上被刷一套房。所以，想使用cf来缓存视频文件。目前，从本地访问速度尚可接受，但不知道其他地区如何，一直想解决这个问题，通过cf作为视频的cdn，让全国的用户能有可接受的访问体验，这样，我就不用担心cdn账单爆仓的问题了。

**Comment by XIU2** [comment]
但是你这样做，容易被 Cloudflare 干掉。。。
Cloudflare CDN 的服务条款 2.8 条写了，不允许缓存/服务于视频文件，除非高级套餐什么的。。。

**Comment by nj-dream** [comment]
啊，这样啊，多谢提醒。我去看看这个条款。我到不介意付费，我就是怕因为被攻击，还需要为攻击者产生的流量支付天价账单。：）

**Comment by nj-dream** [comment]
请问你说的额这个服务条款的url是？我在cf网站下方的条款中，没有找到特别详细的服务条款。。

**Comment by XIU2** [comment]
2.8 Limitation on Serving Non-HTML Content
https://www.cloudflare.com/zh-cn/terms/

**Comment by nj-dream** [comment]
看到了，谢谢！不知道它们的那款方案支持视频。对了，大侠，你知道国内的cdn服务商中，有提供按帐户余额提供服务，余额耗尽就停止服务，而不是继续产生天价账单的服务商吗？我看了一下，好像阿里云、腾讯云，都做不到这一点。

**Comment by XIU2** [comment]
这个你需要发工单问客服。

另外，我不折腾国内，所以不太了解。

**Comment by nj-dream** [comment]
多谢你的帮助，指教了这么多，茅塞顿开，谢谢！
